libraries/stdlib/samples/test/samples/collections/iterables.kt

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright 2010-2017 JetBrains s.r.o.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package samples.collections
+
+import samples.*
+import kotlin.coroutines.experimental.buildIterator
+import kotlin.test.*
+
+class Iterables {
+
+    @Sample
+    fun iterable() {
+        val iterator = Iterable { buildIterator { yieldAll(listOf(1, 2, 3)) } }.iterator()
```

Comment:
Hmm, now it's confusing why do we get an iterator immediately from that `Iterable`: if one wanted to have an iterator, he could use directly that one returned by `buildIterator`.

Usually we want to wrap something in `Iterable` to pass it where `Iterable` is expected. For example there are a lot of extension functions taking `Iterable` as a receiver: `map`, `filter` and so on.

Reply:
Also list of consecutive numbers here can be replaced with a range `1..3`

Reply:
Thanks, fixed.